rest-api/test_backend.py

```diff
@@ -53,3 +22,49 @@ def test_post_delete_leaderboard():
     #Test DELETE request, DELETE a newly-created data to preserve the old content of the database
     resp = app.test_client().delete('/leaderboard', json = return_data)
     assert resp.status_code == 200
+
+def test_post_get_delete_leaderboard():
+    """
+    Test the functionality of the POST, GET and DELETE request
+    """
+
+    # Initial test data to POST
+    data1 = {
+      "name": "Player1",
+      "score": 1000000000
+    }
+    data2 = {
+      "name": "Player2",
+      "score": 34325200
+    }
+    data3 = {
```

Comment:
If testing multiple posts, I suggest adding special cases to test if the leaderboard can handle erroneous data such as negative scores or invalid characters in the name.

Reply:
I realize that our restAPI does not handle erroneous data for POST requests. Therefore, I updated the POST implementation in sample_backend.py to handle such data. Whenever the user attempts to POST a player with an empty name or a negative score, POST will return 412 Error Status Code with message "Invalid data". I also updated test_backend.py to test for these behaviors. You can check these changes under my last commit "Updated sample_backend.py and test_backend.py to handle invalid data for POST requests."